src/modules/sessions-management-pane/management-new-session/management-new-session.tsx

```diff
@@ -35,6 +36,21 @@ export const ManagementNewSession = ({
     setErrors({});
   }, []);
 
+  const setPanel = useContext(SetPanelContext);
+  const agent = useAgent(agentId);
+  const group = useAdminConnection(`/api/groups/${serviceGroupId}`) || {};
+
+  const linkToSettingsHandler = () => {
```

Comment:
separate this function into two functions for Agent and Group. You can create it in jsx because it won`t affect performance